ui/src/views/Tasks/TaskGroup/index.jsx

```diff
@@ -164,16 +169,98 @@ export default class TaskGroup extends Component {
     searchTerm: null,
   };
 
+  static unsubscribe() {
+    if (!TaskGroup.listener) {
+      return;
+    }
+
+    TaskGroup.listener.unsubscribe();
+    TaskGroup.listener = null;
+  }
+
+  static subscribe({ taskGroupId, subscribeToMore }) {
+    if (TaskGroup.listener && TaskGroup.listener.taskGroupId === taskGroupId) {
+      return TaskGroup.listener;
+    }
+
+    if (TaskGroup.listener && TaskGroup.listener.taskGroupId !== taskGroupId) {
+      TaskGroup.unsubscribe();
+    }
+
+    const unsubscribe = subscribeToMore({
+      document: taskGroupSubscription,
+      variables: {
+        taskGroupId,
+        subscriptions: [
+          'tasksDefined',
+          'tasksPending',
+          'tasksRunning',
+          'tasksCompleted',
+          'tasksFailed',
+          'tasksException',
+        ],
+      },
+      updateQuery(previousResult, { subscriptionData }) {
+        const { tasksSubscriptions } = subscriptionData.data;
+        // Make sure data is not from another task group which
+        // can happen when a message is in flight and a user searches for
+        // a different task group.
+        const isFromSameTaskGroupId =
+          tasksSubscriptions.taskGroupId === taskGroupId;
+
+        if (
+          !previousResult ||
+          !previousResult.taskGroup ||
+          !isFromSameTaskGroupId
+        ) {
+          return previousResult;
+        }
+
+        const edges = tasks.has(tasksSubscriptions.taskId)
+          ? previousResult.taskGroup.edges.map(edge => {
+              if (tasksSubscriptions.taskId !== edge.node.status.taskId) {
+                return edge;
+              }
+
+              return dotProp.set(edge, 'node', node =>
+                dotProp.set(node, 'status', status =>
+                  dotProp.set(status, 'state', tasksSubscriptions.state)
+                )
+              );
+            })
+          : tasks.set(tasksSubscriptions.taskId) &&
```

Comment:
I think most readers don't know whether `:` or `&&` has higher precedence.  I don't.  This is probably a good time to use multiple statements instead of stringing together expressions with && in order to achieve sequential execution.  I won't tell the functional-programming police if you don't :)